.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,103 @@
+name: Test & Publish
+
+# Explicit permissions for security
+permissions:
+  contents: read
+  pull-requests: write
+  id-token: write
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test-and-build:
+    name: Test, Lint & Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js 22.15.1
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.15.1
+
+      - name: Get number of CPU cores
+        id: cpu-cores
+        uses: SimenB/github-actions-cpu-cores@v2
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+
+      - uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: |
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+            node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Unit tests
+        run: yarn test --max-workers ${{ steps.cpu-cores.outputs.count }}
+
+      - name: Generate component props
+        run: yarn docs:generate-component-props
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Build packages
+        run: yarn build
+
+  publish:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    needs: [test-and-build]
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    permissions:
+      contents: write # Needed to create releases and tags
```

Comment:
this mentions tagging but I don't see any associated code to create or publish tags. part of the current release workflow is that we tag our releases and push them.

RELEASE.md mentions that a user would publish tags manually. we need another step that does something like :

```
- name: Publish Git Tags
        run: git push --tags
```